src/main/java/io/ballerina/plugins/idea/runconfig/application/BallerinaRunFileAction.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.ballerina.plugins.idea.runconfig.application;
+
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.RunManager;
+import com.intellij.execution.RunnerAndConfigurationSettings;
+import com.intellij.execution.configurations.RunConfiguration;
+import com.intellij.execution.executors.DefaultRunExecutor;
+import com.intellij.execution.runners.ExecutionEnvironmentBuilder;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import io.ballerina.plugins.idea.BallerinaIcons;
+import io.ballerina.plugins.idea.project.BallerinaProjectUtil;
+import io.ballerina.plugins.idea.sdk.BallerinaSdkService;
+import org.jetbrains.annotations.NotNull;
+
+import java.nio.file.Paths;
+import java.util.Objects;
+
+/**
+ * Action for running ballerina application.
+ *
+ * @since 2.0.0
+ */
+public class BallerinaRunFileAction extends AnAction {
+
+    private final String balExtension = ".bal";
```

Comment:
Lets move this to the common constants class, as this will be used across multiple classes